libraries/exoplayer/src/main/java/androidx/media3/exoplayer/drm/KeyLoadInfo.java

```diff
@@ -0,0 +1,57 @@
+package androidx.media3.exoplayer.drm;
+
+import androidx.media3.common.DrmInitData.SchemeData;
+import androidx.media3.exoplayer.source.LoadEventInfo;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Encapsulates info for the sequence of load requests ({@link LoadEventInfo}, which were required
+ * to complete loading a DRM key
+ */
+public class KeyLoadInfo {
+
+  public static class Builder {
+    private LoadEventInfo loadEventInfo;
+    private final List<LoadEventInfo> retriedLoadRequests;
+    private final List<SchemeData> schemeDatas;
+
+    public Builder(List<SchemeData> schemeDatas) {
+      this.schemeDatas = schemeDatas;
+      retriedLoadRequests = new ArrayList<>();
+      loadEventInfo = null;
+    }
+
+    public Builder setMainLoadRequest(LoadEventInfo loadEventInfo) {
+      this.loadEventInfo = loadEventInfo;
+      return this;
+    }
+
+    public Builder addRetryLoadRequest(LoadEventInfo loadEventInfo) {
+      retriedLoadRequests.add(loadEventInfo);
+      return this;
+    }
+
+    public KeyLoadInfo build() {
+      return new KeyLoadInfo(this);
+    }
+  }
+
+  /** The {@link LoadEventInfo} for the initial request to laod the key, or null if no load required
+   */
+  public final LoadEventInfo loadEventInfo;
+
+  /** If the load required multiple retries, the {@link LoadEventInfo} for each retry
+   */
+  public final List<LoadEventInfo> retriedLoadRequests;
+
+  /** The DRM {@link SchemeData} that identifes the loaded key
+   */
+  public final List<SchemeData> schemeDatas;
+
+  public KeyLoadInfo(Builder builder) {
```

Comment:
stylistic/consistency nit: Make this private, and add an explicit `public Builder buildUpon()` method if needed (though I don't think you do need it?)

Reply:
Again, made it private in the renamed class.   Sorry for this, should have fixed it first then renamed rather than rename then fix.